src/encoded/static/components/matrix_target.js

```diff
@@ -0,0 +1,776 @@
+        }
+
+        const count = getSearchResultCount(targetData);
+        PubSub.publish(SEARCH_RESULT_COUNT_PUBSUB, count);
+        this.setState({ targetData });
+    }
+
+    /**
+     * Called when the user scrolls the matrix horizontally within its div to handle scroll
+     * indicators.
+     * @param {object} e React synthetic scroll event
+     */
+    handleOnScroll(e) {
+        this.handleScrollIndicator(e.target);
+    }
+
+    /**
+     * Show a scroll indicator depending on current scrolled position.
+     * @param {object} element DOM element to apply shading to
+     */
+    handleScrollIndicator(element) {
+        if (element) {
+            // Have to use a "roughly equal to" test because of an MS Edge bug mentioned here:
+            // https://stackoverflow.com/questions/30900154/workaround-for-issue-with-ie-scrollwidth
+            const scrollDiff = Math.abs((element.scrollWidth - element.scrollLeft) - element.clientWidth);
+            if (scrollDiff < 2 && !this.state.scrolledRight) {
+                // Right edge of matrix scrolled into view.
+                this.setState({ scrolledRight: true });
+            } else if (scrollDiff >= 2 && this.state.scrolledRight) {
+                // Right edge of matrix scrolled out of view.
+                this.setState({ scrolledRight: false });
+            }
+        } else if (!this.state.scrolledRight) {
+            this.setState({ scrolledRight: true });
+        }
+    }
+
+    render() {
+        const { context } = this.props;
+        const { scrolledRight, targetData, showOrganismRequest, selectedSubTab } = this.state;
+
+        // The server does not return all of this data. So it is hard-coded to make it ever-available.
+        const headers = [
+            {
+                organismName: 'Homo sapiens',
+                assayTitle: 'Histone ChIP-seq',
+                title: 'Homo sapiens | Histone ChIP-seq',
+                url: 'replicates.library.biosample.donor.organism.scientific_name=Homo sapiens&assay_title=Histone ChIP-seq',
+            }, {
+                organismName: 'Homo sapiens',
+                assayTitle: 'TF ChIP-seq',
+                title: 'Homo sapiens | TF ChIP-seq',
+                url: 'replicates.library.biosample.donor.organism.scientific_name=Homo sapiens&assay_title=TF ChIP-seq',
+            }, {
+                organismName: 'Mus musculus',
+                assayTitle: 'Histone ChIP-seq',
+                title: 'Mus musculus | Histone ChIP-seq',
+                url: 'replicates.library.biosample.donor.organism.scientific_name=Mus musculus&assay_title=Histone ChIP-seq',
+            }, {
+                organismName: 'Mus musculus',
+                assayTitle: 'TF ChIP-seq',
+                title: 'Mus musculus | TF ChIP-seq',
+                url: 'replicates.library.biosample.donor.organism.scientific_name=Mus musculus&assay_title=TF ChIP-seq',
+            },
+        ];
```

Comment:
You could put `header` in the global space — it’s static anyway, and in the global space this would get assigned only on page load, not on every render.

Reply:
Move to outside of a class.